lib/Runtime/Types/TypePropertyCache.cpp

```diff
@@ -172,7 +172,9 @@ namespace Js
                 &propertyIndex,
                 &isInlineSlot,
                 &isMissing,
-                &prototypeObjectWithProperty))
+                &prototypeObjectWithProperty)
+            || (prototypeObjectWithProperty != nullptr
+                && prototypeObjectWithProperty != propertyObject))
```

Comment:
Huh? prototypeObjectWithProperty should never by the propertyObject itself.   Otherwise, we will not hit the "proto" inline cache and have prototypeObjectWithProperty not null